server.js

```diff
@@ -29,17 +34,19 @@ let locales = ["sv-SE", "en-US"];
 class Metadata {
     constructor(file, cb) {
         let self = this;
-        this.cb = cb;  
+        this.cb = cb;
         this.db = {};
         this.last_updated = new Date();
         this.count = 0;
-        this.builder = new lunr.Builder();
-        this.builder.pipeline.remove(lunr.trimmer);
-        this.builder.field('title');
-        this.builder.field('tags');
-        this.builder.field('scopes');
 
-        self._p = new Chain([fs.createReadStream(file),parser(),new StreamArray(),data => {
+        if (args.redis) {
+            this.idx = new redisIndexer();
+            this.idx.create();
+        } else {
+            this.idx = new lunrIndexer();
+        };
```

Comment:
Just to satisfy my OCD it would be nice if all the indexer backends implemented a create method (or init or whatever) and lunrIndexer just had a no-op as create(). That way in the future we could switch to a pure table-driven backend initialization and the if-statement could disappear entirely.

Reply:
I understand. Let me change this as well.